compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala

```diff
@@ -88,58 +89,78 @@ final class Conversions(
 
       case SError.SRequiresOnLedger(operation) => setCrash(operation)
 
-      case SError.DamlEMatchError(reason) => setCrash(reason)
-
-      case SError.DamlEUnhandledException(excep) =>
-        builder.setUnhandledException(convertValue(excep.value.toValue))
-
-      case SError.DamlEUserError(msg) => builder.setUserError(msg)
-
-      case SError.DamlETransactionError(reason) =>
-        setCrash(reason)
-
-      case SError.DamlETemplatePreconditionViolated(tid, optLoc, arg) =>
-        val uepvBuilder = proto.ScenarioError.TemplatePreconditionViolated.newBuilder
-        optLoc.map(convertLocation).foreach(uepvBuilder.setLocation)
-        builder.setTemplatePrecondViolated(
-          uepvBuilder
-            .setTemplateId(convertIdentifier(tid))
-            .setArg(convertValue(arg))
-            .build
-        )
-      case SError.DamlELocalContractNotActive(coid, tid, consumedBy) =>
-        builder.setUpdateLocalContractNotActive(
-          proto.ScenarioError.ContractNotActive.newBuilder
-            .setContractRef(mkContractRef(coid, tid))
-            .setConsumedBy(proto.NodeId.newBuilder.setId(consumedBy.toString).build)
-            .build
-        )
-      case SError.DamlEDuplicateContractKey(key) =>
-        builder.setScenarioCommitError(
-          proto.CommitError.newBuilder.setUniqueKeyViolation(convertGlobalKey(key)).build
-        )
-      case SError.DamlEFailedAuthorization(nid, fa) =>
-        builder.setScenarioCommitError(
-          proto.CommitError.newBuilder
-            .setFailedAuthorizations(convertFailedAuthorization(nid, fa))
-            .build
-        )
-
-      case SError.DamlECreateEmptyContractKeyMaintainers(tid, arg, key) =>
-        builder.setCreateEmptyContractKeyMaintainers(
-          proto.ScenarioError.CreateEmptyContractKeyMaintainers.newBuilder
-            .setArg(convertValue(arg))
-            .setTemplateId(convertIdentifier(tid))
-            .setKey(convertValue(key))
-        )
-
-      case SError.DamlEFetchEmptyContractKeyMaintainers(tid, key) =>
-        builder.setFetchEmptyContractKeyMaintainers(
-          proto.ScenarioError.FetchEmptyContractKeyMaintainers.newBuilder
-            .setTemplateId(convertIdentifier(tid))
-            .setKey(convertValue(key))
-        )
+      case SError.SErrorDamlException(interpretationError) =>
+        import interpretation.Error._
+        interpretationError match {
+          case UnhandledException(_, value) =>
+            builder.setUnhandledException(convertValue(value))
+          case UserError(msg) =>
+            builder.setUserError(msg)
+          case ContractNotFound(cid) =>
+            // TODO https://github.com/digital-asset/daml/issues/9974
+            //   we should probably use ${cid.coid} instead of $cid
+            setCrash(s"contract $cid not found")
```

Comment:
Maybe add a comment on why this is a crash:
1. You cannot construct a cid yourself in scenarios/daml script.
2. Contract id fetch failures because a contract was archived or what not are turned into more specific errors so we never produce `ContractNotFound`.